### PR TITLE
Fix ast component to be reactive, turn AstRun into db entity

### PR DESCRIPTION
**Pull request by TomasVotruba**

### CODE REVIEW

File: src/Controller/Ast/ProcessAstFormController.php [hunk 1]
```diff
@@ -6,25 +6,20 @@
...
-    ) {
-    }
-
     public function __invoke(AstFormRequest $astFormRequest): RedirectResponse
     {
-        $astRun = new AstRun(Uuid::v4(), $astFormRequest->getPhpContents());
-        $this->astRunRepository->save($astRun);
+        $astRun = new AstRun();
+        $astRun->content = $astFormRequest->getPhpContents();
+        $astRun->hash = sha1($astFormRequest->getPhpContents());
```
Comment (samsonasik) [inline src/Controller/Ast/ProcessAstFormController.php:18]: The route validation is using uuid, so it should use uuid as well

https://github.com/rectorphp/getrector-com/blob/c6e5a95fce227082f6f345d05de941c55853073a/routes/web.php#L60

Comment (TomasVotruba) [reply src/Controller/Ast/ProcessAstFormController.php:18]: Updated to "hash" :+1:

**Comment by TomasVotruba** [comment]
Ready for review now :+1:

**Comment by samsonasik** [comment]
I got internal server error on submit on /ast page when click "Show AST" button

Screenshot 2024-06-19 at 14 23 03

File: src/Controller/Ast/ProcessAstFormController.php [hunk 2]
```diff
@@ -7,24 +7,19 @@
...
-    public function __construct(
-        private AstRunRepository $astRunRepository
-    ) {
-    }
-
     public function __invoke(AstFormRequest $astFormRequest): RedirectResponse
     {
-        $astRun = new AstRun(Uuid::v4(), $astFormRequest->getPhpContents());
-        $this->astRunRepository->save($astRun);
+        $astRun = new AstRun();
```
Comment (samsonasik) [inline src/Controller/Ast/ProcessAstFormController.php:16]: `AstRun` require 2 parameters in its `__construct()` method, it cause error:

```
Too few arguments to function Rector\Website\Entity\AstRun::__construct(), 0 passed in /Users/samsonasik/www/getrector.org/src/Controller/Ast/ProcessAstFormController.php on line 16 and exactly 2 expected
```

File: src/Controller/Ast/ProcessAstFormController.php [hunk 3]
```diff
@@ -7,24 +7,19 @@
 use Illuminate\Http\RedirectResponse;
 use Illuminate\Routing\Controller;
 use Rector\Website\Entity\AstRun;
```
Comment (samsonasik) [inline src/Controller/Ast/ProcessAstFormController.php:9]: it seems it should be:

```suggestion
use Rector\Website\Ast\Entity\AstRun;
```

**Review by samsonasik — CHANGES_REQUESTED** [review]
There are duplicated `AstRun` classes:

- `Rector\Website\Entity\AstRun`
- `Rector\Website\Ast\Entity\AstRun`

the old one seems should be removed so only use new one which extends `Model`

**Comment by TomasVotruba** [comment]
Indeed, wrong rebase. Fixed now :+1:

**Comment by TomasVotruba** [comment]
Ready now :)

**Comment by samsonasik** [comment]
I am on mobile now, the list to test:

- add code
- click "Show AST" button
- force page refresh
- click node target - get detail node target
- force page refresh: still in detail node target

**Review by samsonasik — CHANGES_REQUESTED** [review]
I tested locally, the "What class to put into Rector::getNodeTypes() method to hook into? " block should keep be styled like previous block:

Screenshot 2024-06-19 at 18 15 00

**Comment by samsonasik** [comment]
nvm, I've fixed it ;)

**Review by samsonasik — APPROVED** [review]

**Comment by samsonasik** [comment]
Let's give it a try, thank you @TomasVotruba

**Comment by samsonasik** [comment]
I deployed, I found a bug that the code pasted moved to first line:

Screenshot 2024-06-19 at 18 32 54

which should be next line after `<?php`

**Comment by samsonasik** [comment]
I created PR :

- https://github.com/rectorphp/getrector-com/pull/2341

for it.

**Pull request merged**